src/soter/boringssl/soter_sym.c

```diff
@@ -99,23 +99,28 @@ soter_sym_ctx_t* soter_sym_ctx_init(const uint32_t alg,
 				    const void* iv,
 				    const size_t iv_length,
 				    bool encrypt){
+  const EVP_CIPHER* evp=algid_to_evp(alg);
+  SOTER_CHECK_PARAM_(evp!=NULL);
   SOTER_CHECK_PARAM_(key!=NULL);
   SOTER_CHECK_PARAM_(key_length!=0);
+  if(salt==NULL){
+    SOTER_CHECK_PARAM_(salt_length==0);
+  }
+  if(iv!=NULL){
+    SOTER_CHECK_PARAM_(iv_length>=EVP_CIPHER_iv_length(evp));
```

Comment:
@secumod would you mind to have a look?

Reply:
checked that we don't use iv_length=0 anywhere in themis code and length defined with values `16`(aes+ctr)/`12`(aes+gcm) that depends of algorithm. and all values of iv we generate in themis code manually and don't rely on crypto backend behaviour

Reply:
This check catches incorrect IV length values which can happen to reach here when the message content has been corrupted. Ideally we would not need them, but crypto backend just expects the length to be correct so we'd better double-check that.

IV length in Themis seems to be defined by `THEMIS_SYM_IV_LENGTH` and Themis simply uses this constant. I imagine that instead of `EVP_CIPHER_iv_length()` we could define some new Soter-specific function similar to `algid_to_evp()` that will return IV length for Soter algorithm. However, relying on crypto backend's expectations seems to be more robust.